### PR TITLE
Use generic gRPC encapsulation rather than explicit protos in Remote Attestation Proxy

### PR DESCRIPTION
**Pull request by conradgrobler**
The Remote Attestation Proxy instance currently has an explicit dependency on the service/proto definitions of the backend (TensorFlow Serving). To make it more generic and reusable this change moves it to a model where it receives a generic gRPC encapsulation request and just passes the bytes through to the backend.

### CODE REVIEW

**Review by tiziano88 — APPROVED** [review]
Very nice!

File: experimental/tf_proxy/client/src/main.rs [hunk 1]
```diff
@@ -24,7 +24,10 @@ use maplit::hashmap;
 use oak_functions_abi::proto::{ConfigurationInfo, Request};
```
Comment (rbehjati) [inline experimental/tf_proxy/client/src/main.rs:24]: pre-existing: is it possible to move the imported parts from `oak_functions_abi` to `grpc_attestation`, and use that both here and in `oak_functions`?

Comment (conradgrobler) [reply experimental/tf_proxy/client/src/main.rs:24]: I created #2495 to track this. I think there are other parts in the crate (and even in `abi.proto`) that is unrelated to the Wasm ABI.

I will do some of the cleanup as part of the PR that will be updating the Oak Functions loader to use the remote attestation proxy.

Comment (rbehjati) [reply experimental/tf_proxy/client/src/main.rs:24]: Sounds good. I think it make sense to do some refactoring and even splitting the `abi.proto` file.

**Pull request merged**